### DISCUSSION TITLE
Guidance with coordinate transforms  and/or unable to reproduce cattle headcount layer

### DISCUSSION THREAD

**Discussion by jamesgerber** [wri/global-pasture-watch#9] [Q&A]
Hi there.  I'm unable to combine the cattle density and the suitable land layers to reproduce the cattle headcount layer.

I suspect the problem is that to manipulate them I'm transforming the CRS, and I'm not doing a good job of it.   

What I expect:  Cattle density * proportion suitable land * area/grid cell = head cattle/grid cell ... but when I try, it doesn't work.  Here are three things I might be doing wrong:
(1) misunderstanding the data layers.  
(2) misunderstanding the appropriateness of trying to beat this into my favorite format (EPSG:4326, lat/long grid) 
(3) my gdalwarp is bad.  Here's what I did:
gdalwarp -te -180 -90 180 90 -tr 0.00833333 0.00833333 -t_srs EPSG:4326  -r average -overwrite gpw_livestock.pot.land_grassland.cropland.rules_p_1km_20220101_20221231_go_epsg.4326_v1.tif potlandgrassland_2022_30sec.tif

gdalwarp -te -180 -90 180 90 -tr 0.00833333 0.00833333 -t_srs EPSG:4326 -r average -overwrite gpw_cattle.density_rf_m_1km_s_20200101_20201231_go_esri.54052_v1.tif gpw_cattledensity_2020_30sec.tif

When I pull these files into Matlab  and sum over (Cattle density * proportion suitable land * area/grid cell) I find 24 billion head of cattle.

Some notes:
the proportion suitable land file (gpw_livestock.pot.land_grassland.cropland.rules_p_1km_20220101_20221231_go_epsg.4326_v1.tif) has a few values just about 100 ... I interpreted this file as percentage of land - so either I'm wrong in that interpretation or there is a bit of noise in your process.

Similarly, there was some file with no data value = -32000 ... that file has just a few negative values.

Also ... one more bit of information: my eventual goal is to get a 30second map of all of the pixels around the world where cattle density is above or below a certain threshold.

**Comment by leandroleal** [comment]
Hi @jamesgerber 

Thanks for exploring the dataset!

For consistency, all modeling steps run in equal-area coordinate system ([Interrupted Goode Homolosine](https://epsg.io/54052)), so pixel size is effectively 1 km². Resulting layer of (`Cattle density * proportion suitable land * 1 km2`) is available in [Zenodo](https://zenodo.org/records/17486471) (Ex: `gpw_cattle.headcount.raw_rf_m_1km_s_20100101_20101231_go_esri.54052_v1.tif`), but these raw output predictions require calibration.

In the paper, we calibrated it linearly adjusting headcount to match with national FAOSTAT statistics (see [Eq.6 in the paper](https://peerj.com/articles/21494/#eqn-6)). Since the calibrated data represent absolute values per 1 km², we had to run a spatial redistribution approach to correct for projection-induced distortions and keep consistent counts at national scale (see [Github](https://github.com/wri/global-pasture-watch/blob/main/gld-1km/workflow/08-reprojection_to_wgs84.py)). Finally, this data was ingested in GEE (see [GPW Annual livestock headcount FAOSTAT-adjusted v1](https://developers.google.com/earth-engine/datasets/catalog/projects_global-pasture-watch_assets_gld-1km_v1_livestock-headcount-faostat_m)) and is publicly available in the platform in EPSG:4326, Lat/Lon grid.

**Regarding your notes**:

> the proportion suitable land file (gpw_livestock.pot.land_grassland.cropland.rules_p_1km_20220101_20221231_go_epsg.4326_v1.tif) has a few values just about 100 ... I interpreted this file as percentage of land - so either I'm wrong in that interpretation or there is a bit of noise in your process.

You are right, that's the percentage of land.

> Similarly, there was some file with no data value = -32000 ... that file has just a few negative values.

RF model are trained on [Poisson criterion](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html), so we shouldn't have negative values on predictions. Maybe, negative values can appear on the overviews due to no data value = -32000. Do you know maybe which file had a few negative values? 

> Also ... one more bit of information: my eventual goal is to get a 30second map of all of the pixels around the world where cattle density is above or below a certain threshold.

In this case, I would 
1) convert the densities to 30second map (EPSG:4326, lat/long grid), 
2) upload them in GEE, and 
3) mask out the ImageCollection `projects/global-pasture-watch/assets/gld-1km/v1/livestock-headcount-faostat_m` with your target density threshold.

**Comment by leandroleal** [comment]
Currently, FAOSTAT-adjusted headcounts in EPSG:4326 (lat/long grid) are available only in GEE (see [GPW Annual livestock headcount FAOSTAT-adjusted v1](https://developers.google.com/earth-engine/datasets/catalog/projects_global-pasture-watch_assets_gld-1km_v1_livestock-headcount-faostat_m)), but we are considering make them available also via Zenodo.
